pennylane_qiskit/qiskit_device.py

```diff
@@ -267,258 +314,27 @@ def _get_state(self, result):
-            par (List[Any]): parameters of the observable(s)
-
-        Returns:
-            array[float]: an array of size ``(len(wires),)`` containing the
-            eigenvalues of the observable
-        """
-        # the standard observables all share a common eigenbasis {1, -1}
-        # with the Pauli-Z gate/computational basis measurement
-        standard_observables = {"PauliX", "PauliY", "PauliZ", "Hadamard"}
-
-        # observable should be Z^{\otimes n}
-        eigvals = pauli_eigs(len(wires))
-
-        if isinstance(observable, list):
-            # tensor product of observables
-
-            # check if there are any non-standard observables (such as Identity, Hadamard)
-            if set(observable) - standard_observables:
-                # Tensor product of observables contains a mixture
-                # of standard and non-standard observables
-                eigvals = np.array([1])
-
-                # group the observables into subgroups, depending on whether
-                # they are in the standard observables or not.
-                for k, g in itertools.groupby(
-                    zip(observable, wires, par), lambda x: x[0] in standard_observables
-                ):
-                    if k:
-                        # Subgroup g contains only standard observables.
-                        # Determine the size of the subgroup, by transposing
-                        # the list, flattening it, and determining the length.
-                        n = len([w for sublist in list(zip(*g))[1] for w in sublist])
-                        eigvals = np.kron(eigvals, pauli_eigs(n))
-                    else:
-                        # Subgroup g contains only non-standard observables.
-                        for ns_obs in g:
-                            # loop through all non-standard observables
-                            if ns_obs[0] == "Hermitian":
-                                # Hermitian observable has pre-computed eigenvalues
-                                p = ns_obs[2]
-                                Hkey = tuple(p[0].flatten().tolist())
-                                eigvals = np.kron(eigvals, self._eigs[Hkey]["eigval"])
-
-                            elif ns_obs[0] == "Identity":
-                                # Identity observable has eigenvalues (1, 1)
-                                eigvals = np.kron(eigvals, np.array([1, 1]))
-
-        elif observable == "Hermitian":
-            # single wire Hermitian observable
-            Hkey = tuple(par[0].flatten().tolist())
-            eigvals = self._eigs[Hkey]["eigval"]
-
-        elif observable == "Identity":
-            # single wire identity observable
-            eigvals = np.ones(2 ** len(wires))
-
-        return eigvals
-
-    def reset(self):
-        self._circuit = QuantumCircuit(self._reg, self._creg, name="temp")
-        self._state = None
+        prob = self.marginal_prob(np.abs(self._state) ** 2, wires)
```

Comment:
see comment above re: probability qubit ordering.

Reply:
Right! Good point! The reversal happens back in `_get_state` (as was originally included):
```python
        # reverse qubit order to match PennyLane convention
        return state.reshape([2] * self.num_wires).T.flatten()
```
(This solution makes use of the fact that transposing the tensor with `(2,2,...2)` shape is equivalent to reversing the qubit ordering from the Qiskit convention.)

The function call tree is done the following way: `QiskitDevice.apply` -> `QiskitDevice.run` -> `QiskitDevice._get_state` (only for `state_backends`).

Reply:
Perfect! Thanks @antalszava, that answers my question.